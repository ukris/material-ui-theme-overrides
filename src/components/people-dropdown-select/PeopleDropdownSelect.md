PeopleDropdownSelect Standard Example:
```jsx
<PeopleDropdownSelect 
    people={[
      {
        "src":"https://material-ui.com/static/images/avatar/1.jpg",
        "name":"Remy Sharp"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/2.jpg",
        "name":"Jerko Raminoff"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/3.jpg",
        "name":"Cindy Baker"
      },
      {
        "src":"",
        "name":"Dean Fremington"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/5.jpg",
        "name":"Winko Wiskardin"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/4.jpg",
        "name":"Paula Howard"
      },
      {
        "src":"",
        "name":"Paula Escariante"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/6.jpg",
        "name":"Domino Ramp"
      },
      {
        "src":"https://material-ui.com/static/images/avatar/7.jpg",
        "name":"Sarah Ilighis"
      }
    ]}

    avatarGroupProps={{
        avatars:[
          {
            "src":"https://material-ui.com/static/images/avatar/2.jpg",
            "name":"Jerko Raminoff"
          },
          {
            "src":"",
            "name":"Dean Fremington"
          },
          {
            "src":"https://material-ui.com/static/images/avatar/4.jpg",
            "name":"Paula Howard"
          },
        ],
        shape: 'circle',
        spacing: 'together',
        max: 3,
        classes: {},
        size: 'sm'
    }} 
/>
```