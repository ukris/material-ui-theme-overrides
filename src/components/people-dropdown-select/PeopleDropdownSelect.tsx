import React, { useState } from 'react'
import Checkbox from '@material-ui/core/Checkbox'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import TextField from '@material-ui/core/TextField'
import Autocomplete from '@material-ui/lab/Autocomplete'
import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank'
import CheckBoxIcon from '@material-ui/icons/CheckBox'
import AvatarGroup from '../avatar-group/AvatarGroup'
import Avatar, { AvatarType} from 'components/avatar/Avatar'
import { AvatarGroupProps } from 'components/avatar-group'
import './style.scss'

const icon = <CheckBoxOutlineBlankIcon fontSize="small" />
const checkedIcon = <CheckBoxIcon fontSize="small" />

interface PeopleDropdownSelectTypes {
    /** List of selectable people */
    people: AvatarType[]

    /** AvatarGroup Component Data. "avatars" prop obj array gets rendered as the people selected by default ("preselected") */
    avatarGroupProps: AvatarGroupProps
    setShowEditor: (content:any) => void
}

export default function PeopleDropdownSelect({ people, avatarGroupProps,  setShowEditor }: PeopleDropdownSelectTypes) {
    const { avatars, max, spacing, shape, size, border, Icon, children } = avatarGroupProps
    const preSelectedPeople = avatars
    const [selectedPeople, setSelectedPeople] = useState(preSelectedPeople)

  const preSelecteds = []
  for(let preSelectedPerson of preSelectedPeople) {
      const index = people.findIndex((person: any) => person.name === preSelectedPerson.name && person.src === preSelectedPerson.src)
      if(index !== -1) preSelecteds.push(people[index])
  }

  return (
    <ClickAwayListener onClickAway={()=> setShowEditor(null)}>
      <div onClick={(ev) => ev.stopPropagation()} className="table-people" data-test="table-peopledropdownselect">
        <Autocomplete
            multiple
            id="checkboxes-tags-demo"
            open={true}
            options={people}
            disableCloseOnSelect
            defaultValue={preSelecteds}
            getOptionLabel={(option: any) => option.name || ''}
            renderOption={(option, { selected }) => {
            return (
                <>
                    <Checkbox
                        icon={icon}
                        checkedIcon={checkedIcon}
                        checked={selected}
                    />
                    <Avatar
                        src={option.src}
                        name={option.name}
                        size={size}
                        border
                    />
                    <span       className="table-people-checkbox-name">{option.name || ''}</span>
                </>
            )
        }}
        style={{ width: 250 }}
        onChange={(event, value) => setSelectedPeople(value)}
        renderInput={(params) => (
            <>
                <div className={`selected ${size}`}>
                    <AvatarGroup
                        avatars={selectedPeople}
                        max={max}
                        spacing={4}
                        size={size}
                        shape={shape}
                        Icon={Icon}
                        border={border}
                        children={children}
                        style={{zIndex: 0}}
                    />
                </div>
                <TextField {...params} variant="outlined"></TextField>
            </>
        )}
        />
      </div>
    </ClickAwayListener>
  )
}

