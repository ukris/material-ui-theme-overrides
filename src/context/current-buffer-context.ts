import { createContext } from 'react'
import { Buffer } from 'types'

interface ICurrentBufferContextProps {
    currentBuffer?: Buffer[]
}

const CurrentBufferContext  = createContext({} as ICurrentBufferContextProps)

export default CurrentBufferContext