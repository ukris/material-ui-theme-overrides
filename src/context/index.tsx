import CurrentBufferContext from './current-buffer-context'
import TableContext from './table-context'
import GlobalContext from './global-context'
import ProjectContext from './project-context'

export { CurrentBufferContext,  GlobalContext, ProjectContext, TableContext }