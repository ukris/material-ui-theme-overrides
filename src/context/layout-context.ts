import { createContext } from 'react'

interface ILayoutContext {
    fullScreen?: boolean
    collapsed?: boolean
    miniMenu?: boolean
    setFullScreen: (full:boolean) => void
    setCollapsed: (collapsed: boolean) => void
    setMiniMenu: (mini:boolean) => void
    width: number
    height: number
    setWidth: (width: number) => void
    setHeight: (height: number) => void
};

const LayoutContext = createContext({} as ILayoutContext);

export default LayoutContext;