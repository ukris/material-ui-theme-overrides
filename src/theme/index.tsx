import { themePalette, LIGHT, DARK } from './palette'
export type LIGHTORDARK = 'light' | 'dark'
export { default } from './theme-provider'
export { default as theme } from './theme'
export { default as makePalette,  } from './palette'
export { default as typography } from './typography'
export { default as ThemeContext } from './theme-context'
export { default as ThemeProvider } from './theme-provider'
export { themePalette, LIGHT, DARK }

const Themes = {
  basil: null,
  crane: null,
  material: null,
  materialDark: null,
  pinky: null,
  rally: null,
  reply: null,
}
export type Color = 'primary' | 'success' | 'info' | 'accent' | 'warning' | 'danger' | 'error' | 'default'
export type ColorValue = {
    light: string,
    main: string,
    dark: string,
    contrastText: string
}
export type ColorObj = {
  [key in Color] : ColorValue
}

export type ThemeType = keyof typeof Themes
export type ThemeZoom = {
  theme: ThemeType
  zoom: number
}
export const AVAILABLE_THEMES = Object.keys(Themes) as ThemeType[]
export const unit = 'rem'
export const tablePadding = [1,0]
export const FONT_BASIS = 2