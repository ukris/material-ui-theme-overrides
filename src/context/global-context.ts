import { ThemeZoom }     from 'theme'
import { Buffer, FontSize } from 'types'
import  defaults, { Settings } from '../settings'
import { Cell, CellValue, Column } from 'apps/table/types'

const FULL_SCREEN = 'full_screen'
const THEME_ZOOM = 'theme_zoom'
const SETTINGS = 'settings'
const CURRENT_BUFFER = 'buffer'
const BUFFERS = 'buffers'
class LocalStorageMock {
    store:any={}
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key:string) {
      return this.store[key] || null;
    }
  
    setItem(key:string, value:string) {
      this.store[key] = value.toString();
    }
  
    removeItem(key:string) {
      delete this.store[key];
    }
  };
  if (!global.localStorage) {
      // @ts-ignore
      global.localStorage = new LocalStorageMock()
  }
interface IGlobalContext {
    buffers: Buffer[]
    nextDomainId: number
    nextProjectId: number
    nextBoardId: number
    nextLaneId: number
    settings: Settings
    fullScreen: boolean
    palette?: any
    themeZoom: ThemeZoom,
    fontSize: FontSize,
    setThemeZoom: (themeZoom: ThemeZoom) => void
    setSettings : (settings: Settings) => void
    setFullScreen: (isFullScreen: boolean) => void
    focusedCell?: Cell | null
    currentBuffer?: Buffer
    domainId?: number
    projectId?: number
    boardId?: number
    laneBy?: number
    laneId?: string
    rows: any[]
    columns: Column[]
    anchorEl?: any
    setCurrentBuffer: (buffers: Buffer) => void
    setBuffers: (buffers: Buffer[]) => void
    setDomainId:(domainId: number) => void
    setProjectId:(projectId: number) => void
    setBoardId:(boardId: number) => void
    setLaneId:(laneId: string) => void
    setLaneBy:(labeBy: number) => void
    setRows: (rows:any[]) => void
    setColumns: (columns: Column[]) => void
    getColumn: (idx: number) => Column
    updateCell: (cell: Cell, newVal: CellValue) => void
    setFocusedCell:  (cell: Cell|null) =>void
    setAnchorEl: (el:any) => void
    theme: any
    setTheme: (theme:any)  => void
    /*

    setDomainId: (domainId: number) => void
    setProjectId: (projectId: number) => void
    setBoardId: (boardId: number) => void
    setLaneBy?: (laneBy?: string) => void // split by/ group by colId
    setLaneId: (laneId: string) => void // example  laneby status, laneId = 'FINISHED, 'INPROGRESS' etc 
    setRowIndex: (idx: number) => void
    setColIndex: (idx: number) => void
    setRowId: (id: string) => void
    setColId: (id: string) => void
    setRange: (range: number[]) => void
    */
}

const fullScreen = !!(localStorage && localStorage.getItem(FULL_SCREEN))
const tZoom = localStorage && localStorage.getItem(THEME_ZOOM)
const themeZoom:ThemeZoom = tZoom ? JSON.parse(tZoom) as ThemeZoom : { zoom: 1, theme: 'material'}
const defaultS = localStorage && localStorage.getItem(SETTINGS)
const defaultSettings: Settings = defaultS ? JSON.parse(defaultS) : defaults
let lBuffer = localStorage && localStorage.getItem(CURRENT_BUFFER)
const currentBuffer:Buffer = lBuffer ? JSON.parse(lBuffer) : []
lBuffer = localStorage && localStorage.getItem(BUFFERS)
const buffers:Buffer[] = lBuffer ? JSON.parse(lBuffer) : []

export const GlobalContext: IGlobalContext = {
    buffers,
    currentBuffer,
    nextDomainId: 2,
    nextProjectId: 2,
    nextBoardId: 2,
    nextLaneId: 2,
    fullScreen,
    themeZoom,
    settings: defaultSettings,
    theme: null,
    fontSize: {
        xs: '0.5rem',
        sm: '0.75rem',
        md: '1rem',
        lg: '1.25rem',
        xl: '1.5rem'
    },
    columns: [],
    rows: [],
    setFocusedCell: (cell: Cell|null)  => {
        GlobalContext.focusedCell = cell;
    },
    setThemeZoom: (themeZoom:ThemeZoom) => {
        (async() => localStorage && localStorage.setItem(THEME_ZOOM, 
            JSON.stringify(themeZoom)))()
        GlobalContext.themeZoom = themeZoom
    },
    setSettings: (settings: Settings) => {
        (async() => localStorage  && localStorage.setItem(SETTINGS, 
            JSON.stringify(settings)))()
        GlobalContext.settings = settings
    },
    setCurrentBuffer: (buffer: Buffer) => {
        (async() => localStorage  && localStorage.setItem(CURRENT_BUFFER, 
            JSON.stringify(buffer)))()
        GlobalContext.currentBuffer = buffer
    },
    setBuffers: (buffers: Buffer[]) => {
        (async() => localStorage  && localStorage.setItem(BUFFERS, 
            JSON.stringify(buffers)))()
        GlobalContext.buffers = buffers
    },
    setFullScreen: (screen: boolean) => {
        (async() => localStorage  && localStorage.setItem(FULL_SCREEN, screen.toString()))()
        GlobalContext.fullScreen = screen
    },
    setDomainId: (domainId: number) => {
        GlobalContext.domainId = domainId
    },
    setProjectId: (projectId: number) => {
        GlobalContext.projectId = projectId
    },
    setBoardId: (boardId: number) => {
        GlobalContext.boardId = boardId
    },
    setLaneId:(laneId: string) => {
        GlobalContext.laneId = laneId
    },
    setLaneBy:(laneBy: number) => {
        GlobalContext.laneBy = laneBy
    },
    setRows: (rows: any[]) => {
        GlobalContext.rows = rows
    },
    setColumns: (columns: Column[]) => {
        GlobalContext.columns = columns
    },
    getColumn: (idx:number) => GlobalContext.columns[idx],
    updateCell: (cell: Cell, newVal: CellValue) => {
        const { rowIndex, colIndex } = cell
        GlobalContext.rows[rowIndex][colIndex] = newVal
    },
    setAnchorEl: (el:any) => {
        GlobalContext.anchorEl = el
    },
    setTheme: (theme: any) => {
        GlobalContext.theme = theme
    }
}

export default GlobalContext