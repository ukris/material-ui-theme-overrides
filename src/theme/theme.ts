import { createMuiTheme, responsiveFontSizes} from '@material-ui/core'
import PaletteOptions from '@material-ui/core/styles/createMuiTheme';
import { GlobalContext } from 'context/global-context'
import { FONT_BASIS, makePalette, ThemeZoom, unit, tablePadding } from './index'
import { roundNum } from 'helpers'
// https://www.sipios.com/blog-tech/how-to-use-styled-components-with-material-ui-in-a-react-app
import typography from './typography'

function createTheme(variant: ThemeZoom):any {
  let palette:any = makePalette(variant)
  const { zoom } = variant
  const { colors } = palette
  delete palette.colors
  GlobalContext.palette = colors
  GlobalContext.palette.type = palette.type
  GlobalContext.palette.contrastText = palette.contrastText
  palette = {...palette, ...colors}
  GlobalContext.setThemeZoom(variant)
  const theme = createMuiTheme({
    zoomFontSize: (factor:number):string => `${roundNum(0.30 * factor * zoom)}${unit}`,
    zoomSpacing: (factor: number) => spacing(roundNum(factor * zoom)),
    fontSize: GlobalContext.fontSize,
    spacing: factor => `${roundNum(0.60 * factor)}${unit}`,
    lineHeight: GlobalContext.fontSize
  });
  const { spacing, zoomSpacing, zoomFontSize } = theme
  const fontSize =  {
      xs: zoomFontSize(FONT_BASIS),
      sm: zoomFontSize(FONT_BASIS+1),
      md: zoomFontSize(FONT_BASIS+2),
      lg: zoomFontSize(FONT_BASIS+3),
      xl: zoomFontSize(FONT_BASIS+4)
  }
  const lineHeight =  {
    xs: zoomFontSize(FONT_BASIS+0.5),
    sm: zoomFontSize(FONT_BASIS+1.5),
    md: zoomFontSize(FONT_BASIS+2.5),
    lg: zoomFontSize(FONT_BASIS+3.5),
    xl: zoomFontSize(FONT_BASIS+4.5)
  }
  let padding:string = tablePadding[0].toString()
  if (tablePadding[0]) {
    padding+= unit
  }
  padding+= ' '
  padding+= tablePadding[1].toString()
  if (tablePadding[1]) {
    padding+= unit
  }
  GlobalContext.fontSize = fontSize
  const modTheme:any = createMuiTheme(
    /**
     * @see https://material-ui.com/customization/themes/#theme-configuration-variables
     * https://material-ui.com/customization/typography/
     */
    {
      palette,
      zoomFontSize,
      typography,
      spacing,
      zoomSpacing,
      fontSize,
      lineHeight,
      props: {
        MuiButtonBase: {
          disableRipple: true
        },
        MuiButton: {
            variant: 'contained',
            disableElevation: true,
        },
        MuiTextField: {
            variant: 'outlined'
        }
      },
      zIndex: {
        appBar: 1200,
        drawer: 1100,
      },
      overrides: {
        MuiDrawer: {
          paper: {
            background: '#18202c',
          },
        },
        MuiPaper: {
          // Name of the component ⚛️ / style sheet
          elevation0: {
            // Name of the rule
            boxShadow: '0 0 14px 0 rgba(53,64,82,.05)', // Some CSS
          },
        },
        MuiTableCell: {
          root: { 
            borderBottom: '1px solid rgba(224, 224, 224, .5)', 
            lineHeight: lineHeight.md,
            fontSize: fontSize.md,
            backgroundColor: theme.palette.background.default,
          },
        },
        MuiAutocomplete: {
          root: {
            backgroundColor: theme.palette.background.default,
            color: theme.palette.getContrastText(theme.palette.background.default)
          }
        },
        MuiButton: {
          root: {
            color: theme.palette.primary.contrastText,
          }
        },
        MuiTextField: {
          root: {
            variant: 'outlined'
          }
        }
    }}
  )
  //  return responsiveFontSizes(modTheme)
  GlobalContext.theme = theme
  return modTheme
}

export default createTheme