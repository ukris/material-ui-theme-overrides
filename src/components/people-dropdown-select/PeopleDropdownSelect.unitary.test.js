import React from 'react'
import { GlobalContext } from 'context'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PeopleDropdownSelect from './PeopleDropdownSelect'
import { ThemeContext, ThemeProvider } from 'theme'

configure({ adapter: new Adapter() })

const themeZoom = GlobalContext.themeZoom;
const setThemeZoom = () => true

global.document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    commonAncestorContainer: {
      nodeName: "BODY",
      ownerDocument: document,
    },
  })


const preselectedPeople = [
    {
        "src":"https://material-ui.com/static/images/avatar/2.jpg",
        "name":"Jerko Raminoff"
    },
    {
        "src":"",
        "name":"Dean Fremington"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/4.jpg",
        "name":"Paula Howard"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/7.jpg",
        "name":"Sarah Ilighis"
    }
]

const selectablePeople = [
    {
        "src":"https://material-ui.com/static/images/avatar/1.jpg",
        "name":"Remy Sharp"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/2.jpg",
        "name":"Jerko Raminoff"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/3.jpg",
        "name":"Cindy Baker"
    },
    {
        "src":"",
        "name":"Dean Fremington"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/5.jpg",
        "name":"Winko Wiskardin"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/4.jpg",
        "name":"Paula Howard"
    },
    {
        "src":"",
        "name":"Paula Escariante"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/6.jpg",
        "name":"Domino Ramp"
    },
    {
        "src":"https://material-ui.com/static/images/avatar/7.jpg",
        "name":"Sarah Ilighis"
    }
]


describe('PeopleDropdownSelect Component | No max prop', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
        <PeopleDropdownSelect 
            people={selectablePeople}
            avatarGroupProps={{
                avatars: preselectedPeople
            }}
            setShowEditor={() => false}
        />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    const wrapper = mount(component)

    it('renders component correctly', () => {
        const peopleDropdownSelectContainer = wrapper.find('[data-test="table-peopledropdownselect"]')
        const avatarGroupContainer = wrapper.find('[data-test="table-avatargroup"]')

        expect(peopleDropdownSelectContainer.length).toEqual(1)
        expect(avatarGroupContainer.length).toEqual(1)
    });

    it('renders correct amounts of Avatars', () => {
        const imgs = wrapper.find('[data-test="table-avatar"]')

        expect(imgs.length).toEqual(selectablePeople.length + preselectedPeople.length)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})
