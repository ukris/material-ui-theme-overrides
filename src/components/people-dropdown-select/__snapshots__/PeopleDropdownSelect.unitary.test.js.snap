// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleDropdownSelect Component | No max prop matches snapshot 1`] = `
Array [
  " ",
  <div
    className="table-people"
    data-test="table-peopledropdownselect"
    onClick={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      aria-expanded={true}
      aria-owns="checkboxes-tags-demo-popup"
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "width": 250,
        }
      }
    >
      <div
        className="selected undefined"
      >
        <div
          className="AvatarGroup-root-196"
          data-test="table-avatargroup"
          style={
            Object {
              "zIndex": 0,
            }
          }
        >
          <span
            className="Avatar-root-198 Avatar-noClass-202 Avatar-circle-200 Avatar-md-209"
            data-test="table-avatar"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "0",
                "color": "#fff",
                "marginLeft": -4,
                "zIndex": 4,
              }
            }
          >
            JR
          </span>
          <span
            className="Avatar-root-198 Avatar-noClass-202 Avatar-circle-200 Avatar-md-209 Avatar-colorDefault-199"
            data-test="table-avatar"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#707264",
                "color": "#fff",
                "marginLeft": -4,
                "zIndex": 3,
              }
            }
          >
            DF
          </span>
          <span
            className="Avatar-root-198 Avatar-noClass-202 Avatar-circle-200 Avatar-md-209"
            data-test="table-avatar"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "0",
                "color": "#fff",
                "marginLeft": -4,
                "zIndex": 2,
              }
            }
          >
            PH
          </span>
          <span
            className="Avatar-root-198 Avatar-noClass-202 Avatar-circle-200 Avatar-md-209"
            data-test="table-avatar"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#957566",
                "color": "#fff",
                "marginLeft": -4,
                "zIndex": 1,
              }
            }
          >
            SI
          </span>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          onClick={[Function]}
        >
          <div
            className="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
            data-tag-index={0}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiChip-label"
            >
              Jerko Raminoff
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-deleteIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </div>
          <div
            className="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
            data-tag-index={1}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiChip-label"
            >
              Dean Fremington
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-deleteIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </div>
          <div
            className="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
            data-tag-index={2}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiChip-label"
            >
              Paula Howard
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-deleteIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </div>
          <div
            className="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
            data-tag-index={3}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiChip-label"
            >
              Sarah Ilighis
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-deleteIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </div>
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-controls="checkboxes-tags-demo-popup"
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            disabled={false}
            id="checkboxes-tags-demo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Close"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator MuiAutocomplete-popupIndicatorOpen"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Close"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-301 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-302"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
]
`;
